File: Modul-4/Soal-3/PRAK403-2410817210016-MUHAMMADRIZKYAKBAR.c
#include <stdio.h>

void Crossnum(int a, int b) {
    int step = (a < b) ? 1 : -1;
    int x = a, y = b;

    while ((step == 1 && x <= b && y >= a) || (step == -1 && x >= b && y <= a)) {
        printf("%d %d", x, y);
        if ((step == 1 && x + step <= b && y - step >= a) || (step == -1 && x + step >= b && y - step <= a)) {
            printf(" - ");
        }
        x += step;
        y -= step;
    }
    printf("\n");
}

int main() {
    int angka1, angka2;
    printf("Masukkan angka: ");
    scanf("%d %d", &angka1, &angka2);

    Crossnum(angka1, angka2);

    return 0;
}

File: Modul-4/Soal-3/PRAK403-2410817210016-MUHAMMADRIZKYAKBAR.py
def Crossnum(a, b):
    step = 1 if a < b else -1
    x, y = a, b

    hasil = []
    while (step == 1 and x <= b and y >= a) or (step == -1 and x >= b and y <= a):
        hasil.append(f"{x} {y}")
        x += step
        y -= step

    print(" - ".join(hasil))


angka1, angka2 = map(int, input("Masukkan angka: ").split())
Crossnum(angka1, angka2)
